Drivers/Src/Lights.c

```diff
@@ -84,6 +84,34 @@ uint16_t Lights_Bitmap_Read(light_t light) {
     return lightStatesBitmap;
 }
 
+/**
+
+ * @brief   Set light toggling
+ * @param   light Which light to enable toggling for
+ * @param   state State to set toggling
+ * @return  void
+ */
+void Toggle_Set(light_t light, State state) {
+    // Mutex not needed here because only BlinkLights uses this bitmap
+    if(state==ON)
+        lightToggleBitmap |= (0x01<<light);
+    else if(state==OFF)
+        lightToggleBitmap &= ~(0x01<<light);
+}
+
+/**
+ * @brief Toggles a light. Should be used only after Toggle_Enable has been called for this light so that we are accurately tracking the enabled and disabled lights
+ * @param light Which light to toggle
+*/
+void Toggle_Light(light_t light){
+    State lightState = Lights_Read(light);
+    if(lightState == OFF){
+        Lights_Set(light, ON);
+    } else {
+        Lights_Set(light, OFF);
+    }
```

Comment:
This is not thread-safe, since you can get interrupted in between reading and setting the light's state. Not sure if you are relying on this to be thread-safe or not

Reply:
The only task calling this should be BlinkLights, so I think it should be OK